backend/src/main/java/com/karankumar/bookproject/backend/model/Book.java

```diff
@@ -244,6 +246,20 @@ public BookBuilder edition(Integer edition) {
         }
     }
 
+    public void addPublisher(@NonNull Publisher publisher) {
+        Set<Publisher> bookPublishers = this.getPublishers();
+        if (bookPublishers == null) {
+            this.setPublishers(Stream.of(publisher).collect(Collectors.toSet()));
+        } else {
+            bookPublishers.add(publisher);
+            this.setPublishers(bookPublishers);
```

Comment:
Instead of using the setter, you can use the `add()` method for the set to add one element.

We can also add the book for the publisher we're adding here. Pseudocode:

```java
bookPublishers.get(publisher).getBooks().add(this);
```

We can then remove the mapping code in the `Publisher` entity.